language-support/ts/codegen/tests/src/DA/Test/Daml2Ts.hs

```diff
@@ -134,6 +134,37 @@ tests damlTypes yarn damlc daml2ts davl = testGroup "daml2ts tests"
         (exitCode, _, err) <- readProcessWithExitCode daml2ts ([groverDar, superGroverDar] ++ ["-o", daml2tsDir, "-p", here </> "package.json"]) ""
         assertBool "A different names for same package error was expected." (exitCode /= ExitSuccess && isJust (stripInfix "Different names ('grover-1.0' and 'super-grover-1.0') for the same package detected" err))
 
+  , testCaseSteps "Bad package.json test" $ \step -> withTempDir $ \here -> do
```

Comment:
An integration test seems a bit overkill (as opposed to just testing a library function that parses the JSON) but given that you already wrote it let’s just keep it for now.